backend/cms/page_xliff_converter.py

```diff
@@ -0,0 +1,312 @@
+
+            site_default_language = page.site.default_language
+            languages = page.site.languages
+
+            source_page_translation = page.get_translation(site_default_language.code)
+            if not source_page_translation:
+                source_page_translation = page_translations[0]
+            source_page_language = source_page_translation.language
+
+            xliff_files = []
+            for language in languages:
+                if language.code != source_page_language.code:
+                    xliff_files.append(self.export_page_translation_xliff(source_page_translation, language.code))
+
+            zip_file_name = "page_{0}.zip".format(page.id)
+            zip_file_path = os.path.join(XLIFFS_DIR, 'pages', str(uuid.uuid4()), zip_file_name)
+            self._create_zip_file(xliff_files, zip_file_path)
+
+            # delete xliff files after created zip file
+            [self._delete_file(xliff_file) for xliff_file in xliff_files]
+
+        return zip_file_path
+
+    @staticmethod
+    def save_page_xliff(page_xliff, user):
+        result = False
+        try:
+            if page_xliff.page_id and page_xliff.title and page_xliff.text and page_xliff.language_code:
+                page = Page.objects.get(id=int(page_xliff.page_id))
+                page_translation = page.get_translation(page_xliff.language_code)
+                if page_translation:
+                    page_translation.title = page_xliff.title
+                    page_translation.text = page_xliff.text
+                elif len(page.languages) > 0:
+                    target_language = None
+                    for language in page.site.languages:
+                        if page_xliff.language_code == language.code:
+                            target_language = language
+                            break
+                    if target_language:
+                        slug = slugify(page_xliff.title)
+                        source_page_translation = list(page.page_translations.all())[0]
+                        page_translation = PageTranslation.objects.create(
+                            slug=slug,
+                            title=page_xliff.title,
+                            text=page_xliff.text,
+                            status=source_page_translation.status,
+                            language=target_language,
+                            public=source_page_translation.public,
+                            page=page,
+                            creator=user
+                        )
+
+                if page_translation:
+                    page_translation.save()
+                    result = True
+
+        except Exception as ex:
+            pass
+
+        return result
+
+    def import_xliff_file(self, file_path, user):
+        result = False
+        if file_path.startswith(XLIFFS_DIR) and file_path.endswith(('.xlf', '.xliff')) and os.path.isfile(file_path):
+            with open(file_path, 'r') as f:
```

Comment:
Same issue as above, in my docker environment the file gets opened in ascii encoding, which fails. Probably we should use  `open(file_path, 'r', encoding='utf-8')` here.

```
======================================================================
ERROR: test_import_xliff_file_translated (cms.tests.PageXliffHelperTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/integreat-cms/backend/cms/tests.py", line 540, in test_import_xliff_file_translated
    result = self.page_xliff_helper.import_xliff_file(file_path, self.user)
  File "/opt/integreat-cms/backend/cms/page_xliff_converter.py", line 292, in import_xliff_file
    xliff_content = f.read()
  File "/usr/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 282: ordinal not in range(128)
```